src/analysis/inner_product_space/gram_schmidt_ortho.lean

```diff
@@ -252,11 +327,84 @@ end
 
 lemma span_gram_schmidt_normed_range (f : ι → E) :
   span 𝕜 (range (gram_schmidt_normed 𝕜 f)) = span 𝕜 (range (gram_schmidt 𝕜 f)) :=
-by simpa only [image_univ.symm] using span_gram_schmidt_normed 𝕜 f univ
-
-/-- When given a basis, `gram_schmidt_normed` produces an orthonormal basis. -/
-noncomputable def gram_schmidt_orthonormal_basis [fintype ι] (b : basis ι 𝕜 E) :
-  orthonormal_basis ι 𝕜 E :=
-orthonormal_basis.mk
-  (gram_schmidt_orthonormal 𝕜 b b.linear_independent)
-  (((span_gram_schmidt_normed_range 𝕜 b).trans (span_gram_schmidt 𝕜 b)).trans b.span_eq).ge
+by simpa only [image_univ.symm] using span_gram_schmidt_normed f univ
+
+section orthonormal_basis
+variables [fintype ι] [finite_dimensional 𝕜 E] (h : finrank 𝕜 E = fintype.card ι) (f : ι → E)
+include h
+
+/-- Given an indexed family `f : ι → E` of vectors in an inner product space `E`, for which the
+size of the index set is the dimension of `E`, produce an orthonormal basis for `E` which agrees
+with the orthonormal set produced by the Gram-Schmidt orthonormalization process on the elements of
+`ι` for which this process gives a nonzero number. -/
+noncomputable def gram_schmidt_orthonormal_basis : orthonormal_basis ι 𝕜 E :=
+((gram_schmidt_orthonormal' f).exists_orthonormal_basis_extension_of_card_eq h).some
+
+lemma gram_schmidt_orthonormal_basis_apply {f : ι → E} {i : ι}
+  (hi : gram_schmidt_normed 𝕜 f i ≠ 0) :
```

Comment:
What's this equal to in the `=0` case? Or is it indeterminate?

Reply:
Can you please transfer this comment to the dependent PR #16477?